plugins/github-actions/src/api/GithubActionsClient.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GithubActionsApi } from './GithubActionsApi';
+import { Build, BuildDetails, BuildStatus, WorkflowRun } from './types';
+import { Entity } from '@backstage/catalog-model';
+
+export class GithubActionsClient implements GithubActionsApi {
+  async listBuilds(entity: Entity, token: Promise<string>): Promise<Build[]> {
+    // ### Feedback request ###
+    // I asumed the following: (maybe not the best. Ideally this should come from the link to the component.yaml file)
+    // entity.metadata.namespace => org name
+    // entity.metadata.name => repo name
+    // entityUri -> entity:spotify:backstage
+
+    let url: string;
+    if (entity.metadata.name !== '') {
+      url = `https://api.github.com/repos/${entity.metadata.namespace}/${entity.metadata.name}/runs`;
+    } else {
+      url = 'https://api.github.com/repos/spotify/backstage/actions/runs';
+    }
+
+    const response = await fetch(url, {
+      headers: new Headers({
+        Authorization: `Bearer ${await token}`,
+      }),
+    });
+
+    if (response.status > 200) {
```

Comment:
2xx should be okay

Reply:
yeah, my bad... should have been > 299 :)